app/components/ReportingRequirement/CollapsibleReport.tsx

```diff
@@ -0,0 +1,122 @@
+import { faChevronDown, faChevronUp } from "@fortawesome/free-solid-svg-icons";
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import StatusBadge from "components/StatusBadge";
+import { DateTime } from "luxon";
+import { useState } from "react";
+import { graphql, useFragment } from "react-relay";
+import { CollapsibleReport_reportingRequirement$key } from "__generated__/CollapsibleReport_reportingRequirement.graphql";
+
+interface Props {
+  title: string;
+  startOpen?: boolean;
+  reportingRequirement: CollapsibleReport_reportingRequirement$key;
+}
+
+const getBadgeForDates = (
+  reportDueDateString: string,
+  submittedDateString: string
+) => {
+  const parsedDueDate =
+    reportDueDateString &&
+    DateTime.fromISO(reportDueDateString, {
+      setZone: true,
+      locale: "en-CA",
+    }).startOf("day");
+  const parsedSubmittedDate =
+    submittedDateString &&
+    DateTime.fromISO(submittedDateString, {
+      setZone: true,
+      locale: "en-CA",
+    }).startOf("day");
+
+  if (parsedDueDate && parsedSubmittedDate) {
+    {
+      return (
+        <StatusBadge
+          variant="complete"
+          label={`Complete (${parsedSubmittedDate.toLocaleString(
+            DateTime.DATE_MED
+          )})`}
+        />
+      );
+    }
+  } else if (parsedDueDate && !parsedSubmittedDate) {
+    const dueIn = parsedDueDate.diff(
+      // Current date without time information
+      DateTime.now().setZone("America/Vancouver").startOf("day"),
+      "days"
+    ).days;
+
+    return dueIn < 0 ? (
+      <StatusBadge variant="late" />
+    ) : (
+      <StatusBadge variant="onTrack" />
+    );
+  } else {
+    return null;
+  }
+};
+
+const CollapsibleReport: React.FC<Props> = ({
+  title,
+  startOpen,
+  reportingRequirement,
+  children,
+}) => {
+  const data = useFragment(
+    graphql`
+      fragment CollapsibleReport_reportingRequirement on ReportingRequirement {
+        reportDueDate
+        submittedDate
+      }
+    `,
+    reportingRequirement
+  );
+
+  // By default, start with the component open if the submittedDate is empty
+  const [isOpen, setIsOpen] = useState(
+    startOpen !== undefined ? startOpen : !data.submittedDate
+  );
+
+  return (
+    <>
+      <div className="reportContainer">
+        <header className="reportHeader" onClick={() => setIsOpen(!isOpen)}>
+          <h3>{title}</h3>
+          <div className="reportStatus">
+            {getBadgeForDates(data.reportDueDate, data.submittedDate)}
+          </div>
+          <div className="toggleIcon">
+            <FontAwesomeIcon icon={isOpen ? faChevronUp : faChevronDown} />
+          </div>
+        </header>
+        {/* Rendering hidden children is necessary in case they contain form elements with validation */}
```

Comment:
It's also better for accessibility and performance (faster to toggle visibility than mount/unmount components)